flask_dance/contrib/orcid.py

```diff
@@ -0,0 +1,97 @@
+from flask import g
+from werkzeug.local import LocalProxy
+
+from flask_dance.consumer import OAuth2ConsumerBlueprint
+
+__maintainer__ = "Matthew Evans <git@ml-evs.science>"
+
+
+def make_orcid_blueprint(
+    client_id=None,
+    client_secret=None,
+    *,
+    scope=None,
+    redirect_url=None,
+    redirect_to=None,
+    login_url=None,
+    authorized_url=None,
+    session_class=None,
+    storage=None,
+    rule_kwargs=None,
+    use_orcid_sandbox=False,
+):
+    """
+    Make a blueprint for authenticating with ORCID (https://orcid.org)
+    using OAuth2.
+
+    This requires a client ID and client secret from ORCID. You should
+    either pass them to this constructor, or make sure that your Flask
+    application config defines them, using the variables
+    :envvar:`ORCID_OAUTH_CLIENT_ID` and :envvar:`ORCID_OAUTH_CLIENT_SECRET`.
+
+    The ORCID Sandbox API (https://sandbox.orcid.org) will be used if
+    the ``use_orcid_sandbox`` argument is set to true.
```

Comment:
I actually think that `sandbox` is a clear enough name. The function is already named `make_orcid_blueprint()`, including "orcid" in the argument name seems unnecessary. Let's keep it concise!

Reply:
Fair, it felt a bit ambiguous to me (given that its really just contacting a different OAuth server) but in context it should be clear enough!